### PR TITLE
[4.4] Noupdate available to joomla 5

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #42034 .

### Summary of Changes
If your joomla 4.4 web site is on a server with a php or db version **below** the minimum requirements for Joomla 5 then we do NOT offer an update.

### Actual result BEFORE applying this Pull Request
No mention at all that its because the site doesnt match the requirements for 5.0. Especially confusing because the dashboard says a new version is available.
image

### Expected result AFTER applying this Pull Request
Clear explanation that you cannot update because of server requirements. Advice to contact your host and link to the requirments

image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
> Warning: Undefined property: Joomla\Component\Joomlaupdate\Administrator\View\Joomlaupdate\HtmlView::$reasonNoDownload in \administrator\components\com_joomlaupdate\tmpl\joomlaupdate\noupdate.php on line 23

File: administrator/language/en-GB/com_joomlaupdate.ini [hunk 1]
```diff
@@ -44,7 +44,8 @@ COM_JOOMLAUPDATE_MINIMUM_STABILITY_STABLE="Stable"
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_APPEND="Upload and Update"
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_BUTTON_ADD="Retry check for update"
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_CONTENT="An update to Joomla %1$s was found, but it wasn't possible to fetch the download URL for that update. Either the update to Joomla %1$s is not available for your stability level or there is a problem with the Joomla Update Server.<br>Please try to download the update package from <a href=\"https://downloads.joomla.org/latest\">the official Joomla download page</a> and use the Upload and Update function."
-COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_TITLE="We can't find a download URL"
+COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_REASON="An update to Joomla %1$s was found but your web server doesn't meet the <a href=\"https://downloads.joomla.org/technical-requirements\" target=\"_blank\" rel=\"noopener noreferrer\"> minimum requirements</a>. Please contact your web host to update your server.<br>"
```
Comment (Quy) [inline administrator/language/en-GB/com_joomlaupdate.ini]:
```suggestion
COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_REASON="An update to Joomla %1$s was found but your web server doesn't meet the <a href=\"https://downloads.joomla.org/technical-requirements\" target=\"_blank\" rel=\"noopener noreferrer\">minimum requirements</a>. Please contact your web host to update your server.<br>"
```

File: administrator/language/en-GB/com_joomlaupdate.ini [hunk 2]
```diff
@@ -44,7 +44,8 @@ COM_JOOMLAUPDATE_MINIMUM_STABILITY_STABLE="Stable"
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_APPEND="Upload and Update"
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_BUTTON_ADD="Retry check for update"
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_CONTENT="An update to Joomla %1$s was found, but it wasn't possible to fetch the download URL for that update. Either the update to Joomla %1$s is not available for your stability level or there is a problem with the Joomla Update Server.<br>Please try to download the update package from <a href=\"https://downloads.joomla.org/latest\">the official Joomla download page</a> and use the Upload and Update function."
-COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_TITLE="We can't find a download URL"
+COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_REASON="An update to Joomla %1$s was found but your web server doesn't meet the <a href=\"https://downloads.joomla.org/technical-requirements\" target=\"_blank\" rel=\"noopener noreferrer\">minimum requirements</a>. Please contact your web host to update your server.<br>"
```
Comment (Quy) [inline administrator/language/en-GB/com_joomlaupdate.ini:47]: I understand the `<br>` , however, should it be concatenated where it is being used?

Comment (brianteeman) [reply administrator/language/en-GB/com_joomlaupdate.ini:47]: If you can find another way to display that text without concat then I am all ears

Comment (Quy) [reply administrator/language/en-GB/com_joomlaupdate.ini:47]: Looking at the code closer, I can't think of a way.

Comment (brianteeman) [reply administrator/language/en-GB/com_joomlaupdate.ini:47]: I did spend quite a while trying to do it within the confines of the existing code but couldnt.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 815e885d900c9dc91e65f416471c6b15640a6d8a<br /><br />Tested successfully in Joomla 4.4.0-rc2-dev of 9 October and Wampserver 3.3.1 with PHP 8.0.26 and 8.2.11.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42103">issues.joomla.org/tracker/joomla-cms/42103</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 815e885d900c9dc91e65f416471c6b15640a6d8a<br /><br />I've tested with an own update site on which I've adjusted the requirements.
- PHP minimum not fulfilled => Ok, new message shown.
Custom URL https://test5.richard-fath.de/next_major_list_test-pr-42103_phpversion.xml
- Database minimum not fulfilled => Ok, new message shown.
Custom URL https://test5.richard-fath.de/next_major_list_test-pr-42103_dbversion.xml
- No update found => Ok, like without this PR.
Default update channel.
- Update found, requirements fulfilled => Ok, like without this PR.
Custom URL https://update.joomla.org/core/nightlies/next_major_list.xml
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42103">issues.joomla.org/tracker/joomla-cms/42103</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42103">issues.joomla.org/tracker/joomla-cms/42103</a>.</sub>

**Comment by brianteeman** [comment]
What is preventing this from being merged? Without it the upgrade experience is attrocious

**Comment by richard67** [comment]
> What is preventing this from being merged? Without it the upgrade experience is attrocious

@brianteeman I don't know. It's a decision by the release managers.

**Comment by MacJoom** [comment]
Thanks - we are looking into it

**Review by tecpromotion — APPROVED** [review]
Thanks all

**Pull request merged**